runtime/src/accounts.rs

```diff
@@ -151,50 +151,65 @@ impl Accounts {
 
             // There is no way to predict what program will execute without an error
             // If a fee can pay for execution then the program will be scheduled
-            let mut accounts: TransactionAccounts = Vec::with_capacity(message.account_keys.len());
+            let mut payer_index = None;
             let mut tx_rent: TransactionRent = 0;
-            for (i, key) in message
+            let mut accounts: Vec<_> = message
                 .account_keys
                 .iter()
                 .enumerate()
-                .filter(|(i, key)| Self::is_non_loader_key(message, key, *i))
-            {
-                let (account, rent) = AccountsDB::load(storage, ancestors, accounts_index, key)
-                    .and_then(|(mut account, _)| {
-                        if message.is_writable(i) && !account.executable {
-                            let rent_due = rent_collector.update(&key, &mut account);
-                            Some((account, rent_due))
-                        } else {
-                            Some((account, 0))
+                .map(|(i, key)| {
+                    if Self::is_non_loader_key(message, key, i) {
+                        if payer_index.is_none() {
+                            payer_index = Some(i);
```

Comment:
@CriesofCarrots we're going to need to figure out how to support this client side

Reply:
We do? Should be no change for the client, right?

Reply:
There are assumptions of fee-payer == keys[0] in several places.  CLI at least to do fee-payer balance checks, Ledger uses it to present the fee-payer account.  Those are the two I'm aware of off hand

Reply:
Huh... the code was always that the first non-program id is the payer. This change doesn't change that. Why did they assume a program id can be payer?

Reply:
In what sense?

Reply:
I think this might be an instance of "standardizing an implementation detail"